### ISSUE TITLE
RStudio App not working

### ISSUE THREAD

**Issue by btsherid** [OSC/Open-OnDemand#74]
Hi,

I have followed the instructions to set up a RStudio App and it successfully starts and I get the button to "Connect to RStudio server", but when I click on the button I just get RStudio Initialization Error, Error occurred during transmission.

In /var/log/messages on the server, I see this:
<pre>
Dec 14 15:25:18 vm-centos7-2-0 rserver[2979]: ERROR system error 2 (No such file or directory) [user-value=btsherid]; OCCURRED AT: rstudio::core::Error rstudio::core::system::user::<unnamed>::userFrom(const rstudio_boost::function<int(T, passwd*, char*, size_t, passwd**)>&, T, rstudio::core::system::user::User*) [with T = const char*] /root/rstudio/src/cpp/core/system/PosixUser.cpp:107; LOGGED FROM: std::string rstudio::server::pam_auth::<unnamed>::userIdentifierToLocalUsername(const std::string&) /root/rstudio/src/cpp/server/ServerPAMAuth.cpp:175
Dec 14 15:25:22 vm-centos7-2-0 rserver[2979]: ERROR system error 13 (Permission denied) [request-uri=/mathjax/MathJax.js?config=TeX-MML-AM_CHTML]; OCCURRED AT: void rstudio::server::session_proxy::<unnamed>::proxyRequest(const rstudio::core::r_util::SessionContext&, rstudio_boost::shared_ptr<rstudio::core::http::AsyncConnection>, const rstudio::core::http::ErrorHandler&, const rstudio::core::http::ConnectionRetryProfile&) /root/rstudio/src/cpp/server/ServerSessionProxy.cpp:577; CAUSED BY: ERROR system error 2 (No such file or directory) [user-value=btsherid]; OCCURRED AT: rstudio::core::Error rstudio::core::system::user::<unnamed>::userFrom(const rstudio_boost::function<int(T, passwd*, char*, size_t, passwd**)>&, T, rstudio::core::system::user::User*) [with T = const char*] /root/rstudio/src/cpp/core/system/PosixUser.cpp:107; LOGGED FROM: void rstudio::server::session_proxy::<unnamed>::logIfNotConnectionTerminated(const rstudio::core::Error&, const rstudio::core::http::Request&) /root/rstudio/src/cpp/server/ServerSessionProxy.cpp:370
Dec 14 15:25:22 vm-centos7-2-0 rserver[2979]: ERROR system error 13 (Permission denied) [request-uri=/rpc/client_init]; OCCURRED AT: void rstudio::server::session_proxy::<unnamed>::proxyRequest(const rstudio::core::r_util::SessionContext&, rstudio_boost::shared_ptr<rstudio::core::http::AsyncConnection>, const rstudio::core::http::ErrorHandler&, const rstudio::core::http::ConnectionRetryProfile&) /root/rstudio/src/cpp/server/ServerSessionProxy.cpp:577; CAUSED BY: ERROR system error 2 (No such file or directory) [user-value=btsherid]; OCCURRED AT: rstudio::core::Error rstudio::core::system::user::<unnamed>::userFrom(const rstudio_boost::function<int(T, passwd*, char*, size_t, passwd**)>&, T, rstudio::core::system::user::User*) [with T = const char*] /root/rstudio/src/cpp/core/system/PosixUser.cpp:107; LOGGED FROM: void rstudio::server::session_proxy::<unnamed>::logIfNotConnectionTerminated(const rstudio::core::Error&, const rstudio::core::http::Request&) /root/rstudio/src/cpp/server/ServerSessionProxy.cpp:370
Dec 14 15:25:23 vm-centos7-2-0 rserver[2979]: ERROR system error 13 (Permission denied) [request-uri=/rpc/client_init]; OCCURRED AT: void rstudio::server::session_proxy::<unnamed>::proxyRequest(const rstudio::core::r_util::SessionContext&, rstudio_boost::shared_ptr<rstudio::core::http::AsyncConnection>, const rstudio::core::http::ErrorHandler&, const rstudio::core::http::ConnectionRetryProfile&) /root/rstudio/src/cpp/server/ServerSessionProxy.cpp:577; CAUSED BY: ERROR system error 2 (No such file or directory) [user-value=btsherid]; OCCURRED AT: rstudio::core::Error rstudio::core::system::user::<unnamed>::userFrom(const rstudio_boost::function<int(T, passwd*, char*, size_t, passwd**)>&, T, rstudio::core::system::user::User*) [with T = const char*] /root/rstudio/src/cpp/core/system/PosixUser.cpp:107; LOGGED FROM: void rstudio::server::session_proxy::<unnamed>::logIfNotConnectionTerminated(const rstudio::core::Error&, const rstudio::core::http::Request&) /root/rstudio/src/cpp/server/ServerSessionProxy.cpp:370
</pre>
Any help getting this working would be appreciated.

Thank You,
Brendan Sheridan

**Comment by btsherid** [comment]
There is something wrong with the singularity container or how it interacts with my cluster node.

If I do the below in the script.sh.erb file, I successfully get logged into RStudio:
rserver --www-port "${port}" --auth-none 0 --auth-pam-helper-path "${RSTUDIO_AUTH}" --auth-encrypt-password 0 --rsession-path "${RSESSION_WRAPPER_FILE}"
#singularity run -B "$TMPDIR:/tmp" "$RSTUDIO_SERVER_IMAGE" \
# --www-port "${port}" \
# --auth-none 0 \
# --auth-pam-helper-path "${RSTUDIO_AUTH}" \
# --auth-encrypt-password 0 \
# --rsession-path "${RSESSION_WRAPPER_FILE}"

-Brendan

**Comment by johrstrom** [comment]
Sounds like you're on the right track.  I'd try to debug this interactively. Get a desktop VNC session going. and try to execute the same `singularity run` command. Or `singularity exec` with sleep command so you can ssh into the same node and try to interactively start rserver with the job's container.

Basically it's going to be an issue with your setup. Maybe the SINGULARITY_BINDPATH, who knows, but you'll likely have to hop in there and see the rstudio logs to find out.

**Comment by btsherid** [comment]
I must be missing something. 

I can run this interactively by setting exporting SINGULARITY_BINDPATH and running the singularity run command and it will run on the same port and use the same password as when launching through OnDemand. I can connect to the server with SSH and X11 and connect to localhost:port and sign into RStudio with that password but I still get Error occurred during transmission. All I get from the interactive output is the same thing I get from the /var/log/messages output I posted earlier. Because the messages are the same I am at a similar loss as to what is causing the issue.

Is there the option to do this without Singularity? It works without but it would only work for a single user. Maybe something hardcoded?

Thanks,
Brendan

**Comment by btsherid** [comment]
I think I've got it working. Following the below two threads I upgraded the rstudio-server RPM on my node, updated the auth script and changed ne to lt, and added --server-data-dir "${TMPDIR}" --secure-cookie-key-file "${TMPDIR}/rstudio-server/secure-cookie-key" to the end of the rserver command.

https://discourse.osc.edu/t/rstudio-without-singularity/773/9
https://discourse.osc.edu/t/rstudio-when-launched-without-singularity-is-having-strange-troubles-with-authentication/1213

I've successfully tested with myself and another user and it also fixed a caching issue I was experiencing previously when running two RStudio apps with different versions of R.

-Brendan
